libcxx/utils/libcxx/sym_check/util.py

```diff
@@ -123,6 +123,40 @@ def adjust_mangled_name(name):
     return name[1:]
 
 
+def extract_object_sizes_from_map(mapfilename: str):
+    maplines = []
+    result = {}
+
+    with open(mapfilename, "r") as f:
+        maplines = f.readlines()
+
+    if len(maplines) == 0:
+        return {}
+
+    # Yes, this is fragile.
```

Comment:
Does this only work with a mapfile generated on Apple platforms, or is that format cross-platform? I haven't played around with mapfiles much in the past, so I'm not very familiar with those.

Reply:
Unfortunately, yes, it only works with the Darwin linker (`/usr/bin/ld`). I attempted to duplicate it with the linker on Linux but, although The GNU Linker does support map files, it does not give us enough information. So, I think that we are stuck with having a solution limited to the Darwin linker.